### ISSUE TITLE
Compressed Bitmap Caching for Facet Data

### ISSUE THREAD

**Issue by roodriigoooo** [kejihela/Elastic-search-with-sBERT-model#4]
At least the implementation I proposed does not really use space efficiently when storing facet data, and memory consumption is not efficient. A compressed caching could reduce the storage space required for facet data, as per [this](https://www.cs.cmu.edu/~ddash/papers/facets-cikm.pdf )paper on Dynamic Faceted Search. 

We could explore libraries or algs that support compressed indexing like Roaring bitmaps.

**Comment by kejihela** [comment]
Yes you are right. The time and space complexity should be put into consideration as well. I will find time to go through the paper prior to the meeting. Thank you.
